ecommerce/api.py

```diff
@@ -387,33 +389,59 @@ def get_new_order_by_reference_number(reference_number):
         raise EcommerceException("Unable to find order {}".format(order_id))
 
 
-def enroll_user_on_success(order):  # pylint: disable=unused-argument
+def complete_order(order):
     """
-    Check if the order is successful, then enroll the user and create local records of their
+    Enrolls a user in all items associated with their Order and gets rid of checkout-related objects
+    so that the user starts fresh next time they go through ecommerce.
+
+    Args:
+        order (Order): A fulfilled order
+    """
+    enroll_user_in_order_items(order)
+
+    # clear the basket
+    with transaction.atomic():
+        BasketItem.objects.filter(basket__user=order.purchaser).delete()
+        CourseRunSelection.objects.filter(basket__user=order.purchaser).delete()
+        CouponSelection.objects.filter(basket__user=order.purchaser).delete()
+
+
+def enroll_user_in_order_items(order):
+    """
+    Enroll the user in the CourseRuns associated with their Order, and create local records of their
     enrollments.
 
     Args:
         order (Order): An order
     """
     basket = order.purchaser.basket
     runs = CourseRun.objects.filter(courserunselection__basket=basket)
-    # pylint: disable=fixme
-    # TODO: Actually enroll the user in the given course runs on Open edX
+    programs = get_order_programs(order)
     company = get_company_affiliation(order)
+
+    try:
+        enroll_in_edx_course_runs(order.purchaser, runs)
+        edx_request_success = True
+    except Exception as exc:  # pylint: disable=broad-except
+        log.exception(
+            "edX enrollment for user %s failed (order id: %s, run ids: %s, response: %s)",
+            order.purchaser.username,
+            order.id,
+            str([run.id for run in runs]),
+            exc.response.content.decode("utf-8"),
```

Comment:
If something other than an HttpError occurs, the exception might not have a response.  Could change to something like:
```python
      exc.response.content.decode("utf-8") if hasattr(exc, "response") else "N/A"
```